redisson/src/main/java/org/redisson/RedissonPermitExpirableSemaphore.java

```diff
@@ -532,49 +565,88 @@ private void unsubscribe(RedissonLockEntry entry) {
 
     @Override
     public String tryAcquire(long waitTime, TimeUnit unit) throws InterruptedException {
-        String res = tryAcquire(1, waitTime, -1, unit);
-        if (res != null && res.startsWith(":")) {
+        List<String> ids = tryAcquire(1, waitTime, -1, unit);
+        if (ids.isEmpty() || hasOnlyNearestTimeout(ids)) {
             return null;
         }
-        return res;
+        return ids.get(0);
     }
 
     @Override
     public void release(String permitId) {
         get(releaseAsync(permitId));
     }
 
+    @Override
+    public int release(List<String> permitsIds) {
+        return get(releaseAsync(permitsIds));
+    }
+
     @Override
     public boolean tryRelease(String permitId) {
         return get(tryReleaseAsync(permitId));
     }
 
+    @Override
+    public int tryRelease(List<String> permitsIds) {
+        return 0;
+    }
+
     @Override
     public RFuture<Boolean> tryReleaseAsync(String permitId) {
         if (permitId == null) {
             throw new IllegalArgumentException("permitId can't be null");
         }
 
-        byte[] id = ByteBufUtil.decodeHexDump(permitId);
```

Comment:
Why isn't the binary form used?

Reply:
That's the part I'm not sure in `cause it can break backwards compatibility.

The reason is that I haven't found a way to parse array of binary identifiers to table/array in Lua script.
And it's needed in [acquire](https://github.com/redisson/redisson/pull/5247/files/a109fe9cf2f453ace715e16333a57cd600091c1a#diff-a6afb637de0f8d40a841263def0f289d6200822972dde04dfed78baf9a77239cR381) and [release](https://github.com/redisson/redisson/pull/5247/files/a109fe9cf2f453ace715e16333a57cd600091c1a#diff-a6afb637de0f8d40a841263def0f289d6200822972dde04dfed78baf9a77239cR633) methods.

If there is a way to unpack it and use in script please advice. Or maybe I can try more complex regex for it